tests/unit/lib/test_runner/test_invoke_testsuite.py

```diff
@@ -0,0 +1,66 @@
+from botocore.exceptions import ClientError, WaiterError
+from unittest import TestCase
+from unittest.mock import Mock
+from samcli.lib.test_runner.invoke_testsuite import SuiteRunner
+from samcli.commands.exceptions import ReservedEnvironmentVariableException
+
+
+class Test_InvokeTestsuite(TestCase):
+    def setUp(self):
+        self.runner = SuiteRunner(None, None)
+        self.params = {
+            "bucket": "test-bucket-name",
+            "path_in_bucket": "test_bucket_path",
+            "ecs_cluster": "test-cluster-name",
+            "container_name": "test-container-name",
+            "task_definition_arn": "test-task-def-arn",
+            "other_env_vars": {},
+            "test_command_options": "--max-fail=1",
+            "subnets": ["subnet-xxxxxxxxxxxxxxxxx"],
+        }
+
+    def test_invalid_var_names(self):
+        self.params["other_env_vars"] = {r"0variable_name": "value", r"\othervariable_name": "othervalue"}
+        with self.assertRaises(ValueError):
+            self.runner.invoke_testsuite(**self.params)
+
+    def test_specify_variable_bucket(self):
```

Comment:
Nit: you can parameterize this test with the next two. Up to you, but calling it out incase you were unaware of test parameterization.

Reply:
I wasn't, but I am now! This is much better, very easy to expand.